packages/conf/conf.go

```diff
@@ -48,78 +55,138 @@ type LogConfig struct {
+
+	PidFilePath           string
+	LockFilePath          string
+	DataDir               string // application work dir (cwd by default)
+	KeysDir               string // place for private keys files: NodePrivateKey, PrivateKey
+	FirstBlockPath        string
+	TLS                   string // TLS is a directory for .well-known and keys. It is required for https
+	MaxPageGenerationTime int64  // in milliseconds
 
 	TCPServer HostPort
 	HTTP      HostPort
-	DB        DBConfig
-	StatsD    StatsDConfig `toml:"-"`
-
-	WorkDir string // application work dir (cwd by default)
-	KeysDir string // place for private keys files: NodePrivateKey, PrivateKey
-
-	Centrifugo CentrifugoConfig
-
-	LogConfig LogConfig
 
+	DB            DBConfig
+	StatsD        StatsDConfig
+	Centrifugo    CentrifugoConfig
+	LogConfig     LogConfig
 	TokenMovement TokenMovementConfig
 }
 
-// Installed web UI installation mode
-var Installed bool
-
 // Config global parameters
-var Config = SavedConfig{
-	StatsD: StatsDConfig{Name: "genesis", HostPort: HostPort{Host: "127.0.0.1", Port: 8125}},
-}
+var Config GlobalConfig
 
-// GetConfigPath returns path from command line arg or default
-func GetConfigPath() string {
-	if *ConfigPath != "" {
-		return *ConfigPath
-	}
-	return filepath.Join(*WorkDirectory, consts.DefaultConfigFile)
-}
-
-// GetPidFile returns path to pid file
-func GetPidFile() string {
-	return filepath.Join(Config.WorkDir, consts.PidFilename)
+// GetPidPath returns path to pid file
+func (c *GlobalConfig) GetPidPath() string {
+	return c.PidFilePath
 }
 
 // LoadConfig from configFile
 // the function has side effect updating global var Config
-func LoadConfig() error {
-	log.WithFields(log.Fields{"path": GetConfigPath()}).Info("Loading config")
-	_, err := toml.DecodeFile(GetConfigPath(), &Config)
-	return err
+func LoadConfig(path string) error {
+	log.WithFields(log.Fields{"path": path}).Info("Loading config")
+
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return errors.Errorf("Unable to load config file %s", path)
+	}
+
+	viper.SetConfigFile(path)
+	err = viper.ReadInConfig()
+	if err != nil {
+		return errors.Wrapf(err, "reading config")
+	}
+
+	err = viper.Unmarshal(&Config)
+	if err != nil {
+		return errors.Wrapf(err, "marshalling config to global struct variable")
+	}
+
+	return nil
 }
 
 // SaveConfig save global parameters to configFile
-func SaveConfig() error {
-	cf, err := os.Create(GetConfigPath())
+func SaveConfig(path string) error {
+	dir := filepath.Dir(path)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		err := os.Mkdir(dir, 0775)
+		if err != nil {
+			return errors.Wrapf(err, "creating dir %s", dir)
+		}
+	}
+
+	cf, err := os.Create(path)
 	if err != nil {
 		log.WithFields(log.Fields{"type": consts.IOError, "error": err}).Error("Create config file failed")
 		return err
 	}
 	defer cf.Close()
-	return toml.NewEncoder(cf).Encode(Config)
+
+	err = toml.NewEncoder(cf).Encode(Config)
+	if err != nil {
+		return err
```

Comment:
logging?

Reply:
This error is logged in the parent function